pennylane/ops/qubit/hamiltonian.py

```diff
@@ -426,6 +426,11 @@ def __repr__(self):
         # Constructor-call-like representation
         return f"<Hamiltonian: terms={qml.math.shape(self.coeffs)[0]}, wires={self.wires.tolist()}>"
 
+    def _ipython_display_(self):
+        # See https://ipython.readthedocs.io/en/stable/config/integrating.html
+        # Displays __str__ in ipython instead of __repr__
```

Comment:
you could probably make this second comment a doc string,

Reply:
This is more of a note to future developers looking at source code, instead of something specifying the API and telling people how to use it in different locations.